fine_tuning_job.go

```diff
@@ -0,0 +1,154 @@
+	Hyperparameters *Hyperparameters `json:"hyperparameters,omitempty"`
+	Suffix          string           `json:"suffix,omitempty"`
+}
+
+type FineTuningJobEventList struct {
+	Object  string          `json:"object"`
+	Data    []FineTuneEvent `json:"data"`
+	HasMore bool            `json:"has_more"`
+}
+
+type FineTuningJobEvent struct {
+	Object    string `json:"object"`
+	ID        string `json:"id"`
+	CreatedAt int    `json:"created_at"`
+	Level     string `json:"level"`
+	Message   string `json:"message"`
+	Data      any    `json:"data"`
+	Type      string `json:"type"`
+}
+
+// CreateFineTuningJob create a fine tuning job.
+func (c *Client) CreateFineTuningJob(
+	ctx context.Context,
+	request FineTuningJobRequest,
+) (response FineTuningJob, err error) {
+	urlSuffix := "/fine_tuning/jobs"
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix), withBody(request))
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// CancelFineTuningJob cancel a fine tuning job.
+func (c *Client) CancelFineTuningJob(ctx context.Context, fineTuningJobID string) (response FineTuningJob, err error) {
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL("/fine_tuning/jobs/"+fineTuningJobID+"/cancel"))
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// RetrieveFineTuningJob retrieve a fine tuning job.
+func (c *Client) RetrieveFineTuningJob(
+	ctx context.Context,
+	fineTuningJobID string,
+) (response FineTuningJob, err error) {
+	urlSuffix := fmt.Sprintf("/fine_tuning/jobs/%s", fineTuningJobID)
+	req, err := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix))
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+type listFineTuningJobEventsParameters struct {
+	after *string
+	limit *int
+}
+
+type ListFineTuningJobEventsParameter func(*listFineTuningJobEventsParameters)
+
+func ListFineTuningJobEventsWithAfter(after string) ListFineTuningJobEventsParameter {
+	return func(args *listFineTuningJobEventsParameters) {
+		args.after = &after
+	}
+}
+
+func ListFineTuningJobEventsWithLimit(limit int) ListFineTuningJobEventsParameter {
+	return func(args *listFineTuningJobEventsParameters) {
+		args.limit = &limit
+	}
+}
+
+// ListFineTuningJobs list fine tuning jobs events.
+func (c *Client) ListFineTuningJobEvents(
+	ctx context.Context,
+	fineTuningJobID string,
+	setters ...ListFineTuningJobEventsParameter,
+) (response FineTuningJobEventList, err error) {
+	parameters := &listFineTuningJobEventsParameters{
+		after: nil,
+		limit: nil,
+	}
+
+	for _, setter := range setters {
+		setter(parameters)
+	}
+
+	requestParamters := ""
+	if parameters.after != nil {
+		requestParamters += fmt.Sprintf("after=%s", *parameters.after)
+	}
+	if parameters.limit != nil {
+		if requestParamters != "" {
+			requestParamters += "&"
+		}
+		requestParamters += fmt.Sprintf("limit=%d", *parameters.limit)
```

Comment:
Could we please use `url.Values` here?

Example from https://pkg.go.dev/net/url#Values.Encode

```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	v := url.Values{}
	v.Add("cat sounds", "meow")
	v.Add("cat sounds", "mew/")
	v.Add("cat sounds", "mau$")
	fmt.Println(v.Encode())

}
```

Reply:
done